src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -175,9 +191,21 @@ private async Task ProcessQueueAsync()
                 try
                 {
                     var (work, activityId, cancellationToken) = queueItem;
+                    CancellationTokenSource? currentWorkCts = null;
                     lspServices = work.LspServices;
 
-                    var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
+                    if (CancelInProgressWorkUponMutatingRequest)
+                    {
+                        // Verify queueItem hasn't already been cancelled before creating a linked
+                        // CancellationTokenSource based on it.
+                        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
can remove this line right?

Reply:
I don't think so. My debugging hit cases where cancellationToken was already cancelled at this line. If that happened, then the CreateLinkedTokenSource call threw an ObjectDisposedException.

Reply:
Sure. Would you prefer a separate try/catch(ObjectDisposedException) tightly around the CreateLinkedTokenSource call (with a potential rethrow of an ObjectCancelledException), or to add an ObjectDisposedException handler to the existing catch (OperationCanceledException ex)?

Reply:
Went ahead and did the ObjectDisposedException handler. Let me know if you would prefer something different.